The job **`aarch64-gnu-llvm-20-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/17909050470/job/50916158629) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/50916158629) [(plain)](https://github.com/rust-lang/rust/commit/f419c93a1011b440998f1f92b89d823204d7f2b5/checks/50916158629/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

---- [codegen] tests/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.rs#aarch64 stdout ----
------FileCheck stdout------------------------------

------FileCheck stderr------------------------------
/checkout/tests/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.rs:67:12: error: CHECK: expected string not found in input
 // CHECK: call <fn_ptr_reify_shim::S as fn_ptr_reify_shim::NotDynCompatible>::not_dyn_name{{$}}}
           ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.aarch64/fn-ptr-reify-shim.ll:73:76: note: scanning from here
; call fn_ptr_reify_shim::DynCompatible::dyn_name_default{{reify.shim.fnptr}}
                                                                           ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.aarch64/fn-ptr-reify-shim.ll:77:3: note: possible intended match here
; call <fn_ptr_reify_shim::S as fn_ptr_reify_shim::NotDynCompatible>::not_dyn_name
  ^

Input file: /checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.aarch64/fn-ptr-reify-shim.ll
Check file: /checkout/tests/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.rs

-dump-input=help explains the following input dump.

Input was:
<<<<<<
            1: ; ModuleID = 'fn_ptr_reify_shim.b6c03fc4bc8896ab-cgu.0' 
            2: source_filename = "fn_ptr_reify_shim.b6c03fc4bc8896ab-cgu.0" 
            3: target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i8:8:32-i16:16:32-i64:64-i128:128-n32:64-S128-Fn32" 
            4: target triple = "aarch64-unknown-none" 
            5:  
            6: @alloc_f5787007e3698542f22dabdab1d42ed6 = private unnamed_addr constant [11 x i8] c"dyn_default", align 1 
            7: @alloc_2172a80514eefbb0a9cb60752ddeb97d = private unnamed_addr constant [15 x i8] c"not_dyn_default", align 1 
            8: @alloc_645829e0bfec2cab47a651c7b4ba9f04 = private unnamed_addr constant [14 x i8] c"dyn_compatible", align 1 
            9: @alloc_d0a58cbf370d44771ccbd4953c6d9fdd = private unnamed_addr constant [18 x i8] c"not_dyn_compatible", align 1 
           10:  
           11: ; fn_ptr_reify_shim::DynCompatible::dyn_name_default{{reify.shim.fnptr}} 
           12: ; Function Attrs: inlinehint noredzone nounwind uwtable 
           13: define internal { ptr, i64 } @"_ZN17fn_ptr_reify_shim13DynCompatible52dyn_name_default$u7b$$u7b$reify.shim.fnptr$u7d$$u7d$17h7670877906a96512E"(ptr align 1 %_1) unnamed_addr #0 !kcfi_type !2 { 
           14: start: 
           15: ; call fn_ptr_reify_shim::DynCompatible::dyn_name_default 
           16:  %0 = call { ptr, i64 } @_ZN17fn_ptr_reify_shim13DynCompatible16dyn_name_default17h73e71e598780e0b1E(ptr align 1 %_1) #2 
           17:  %_0.0 = extractvalue { ptr, i64 } %0, 0 
           18:  %_0.1 = extractvalue { ptr, i64 } %0, 1 
           19:  %1 = insertvalue { ptr, i64 } poison, ptr %_0.0, 0 
           20:  %2 = insertvalue { ptr, i64 } %1, i64 %_0.1, 1 
           21:  ret { ptr, i64 } %2 
           22: } 
           23:  
           24: ; <fn_ptr_reify_shim::S as fn_ptr_reify_shim::DynCompatible>::dyn_name{{reify.shim.fnptr}} 
           25: ; Function Attrs: inlinehint noredzone nounwind uwtable 
           26: define internal { ptr, i64 } @"_ZN73_$LT$fn_ptr_reify_shim..S$u20$as$u20$fn_ptr_reify_shim..DynCompatible$GT$44dyn_name$u7b$$u7b$reify.shim.fnptr$u7d$$u7d$17he6409f97a7efcb24E"(ptr align 1 %_1) unnamed_addr #0 !kcfi_type !2 { 
           27: start: 
           28: ; call <fn_ptr_reify_shim::S as fn_ptr_reify_shim::DynCompatible>::dyn_name 
           29:  %0 = call { ptr, i64 } @"_ZN73_$LT$fn_ptr_reify_shim..S$u20$as$u20$fn_ptr_reify_shim..DynCompatible$GT$8dyn_name17h8a2e331ffcc7d6bbE"(ptr align 1 %_1) #2 
           30:  %_0.0 = extractvalue { ptr, i64 } %0, 0 
           31:  %_0.1 = extractvalue { ptr, i64 } %0, 1 
           32:  %1 = insertvalue { ptr, i64 } poison, ptr %_0.0, 0 
           33:  %2 = insertvalue { ptr, i64 } %1, i64 %_0.1, 1 
           34:  ret { ptr, i64 } %2 
           35: } 
           36:  
           37: ; fn_ptr_reify_shim::DynCompatible::dyn_name_default 
           38: ; Function Attrs: noredzone nounwind uwtable 
           39: define dso_local { ptr, i64 } @_ZN17fn_ptr_reify_shim13DynCompatible16dyn_name_default17h73e71e598780e0b1E(ptr align 1 %self) unnamed_addr #1 !kcfi_type !3 { 
           40: start: 
           41:  ret { ptr, i64 } { ptr @alloc_f5787007e3698542f22dabdab1d42ed6, i64 11 } 
           42: } 
           43:  
           44: ; fn_ptr_reify_shim::NotDynCompatible::not_dyn_name_default 
           45: ; Function Attrs: noredzone nounwind uwtable 
           46: define dso_local { ptr, i64 } @_ZN17fn_ptr_reify_shim16NotDynCompatible20not_dyn_name_default17h63f35d2e3ca4a5fdE() unnamed_addr #1 !kcfi_type !4 { 
           47: start: 
           48:  ret { ptr, i64 } { ptr @alloc_2172a80514eefbb0a9cb60752ddeb97d, i64 15 } 
           49: } 
           50:  
           51: ; <fn_ptr_reify_shim::S as fn_ptr_reify_shim::DynCompatible>::dyn_name 
           52: ; Function Attrs: noredzone nounwind uwtable 
           53: define dso_local { ptr, i64 } @"_ZN73_$LT$fn_ptr_reify_shim..S$u20$as$u20$fn_ptr_reify_shim..DynCompatible$GT$8dyn_name17h8a2e331ffcc7d6bbE"(ptr align 1 %self) unnamed_addr #1 !kcfi_type !3 { 
           54: start: 
           55:  ret { ptr, i64 } { ptr @alloc_645829e0bfec2cab47a651c7b4ba9f04, i64 14 } 
           56: } 
           57:  
           58: ; <fn_ptr_reify_shim::S as fn_ptr_reify_shim::NotDynCompatible>::not_dyn_name 
           59: ; Function Attrs: noredzone nounwind uwtable 
           60: define dso_local { ptr, i64 } @"_ZN76_$LT$fn_ptr_reify_shim..S$u20$as$u20$fn_ptr_reify_shim..NotDynCompatible$GT$12not_dyn_name17h29d9315502a38704E"() unnamed_addr #1 !kcfi_type !4 { 
           61: start: 
           62:  ret { ptr, i64 } { ptr @alloc_d0a58cbf370d44771ccbd4953c6d9fdd, i64 18 } 
           63: } 
           64:  
           65: ; Function Attrs: noredzone nounwind uwtable 
           66: define dso_local void @main() unnamed_addr #1 !kcfi_type !5 { 
           67: start: 
           68:  %_1 = alloca [0 x i8], align 1 
           69: ; call <fn_ptr_reify_shim::S as fn_ptr_reify_shim::DynCompatible>::dyn_name{{reify.shim.fnptr}} 
           70:  %0 = call { ptr, i64 } @"_ZN73_$LT$fn_ptr_reify_shim..S$u20$as$u20$fn_ptr_reify_shim..DynCompatible$GT$44dyn_name$u7b$$u7b$reify.shim.fnptr$u7d$$u7d$17he6409f97a7efcb24E"(ptr align 1 %_1) #2 
           71:  %_unused.0 = extractvalue { ptr, i64 } %0, 0 
           72:  %_unused.1 = extractvalue { ptr, i64 } %0, 1 
           73: ; call fn_ptr_reify_shim::DynCompatible::dyn_name_default{{reify.shim.fnptr}} 
check:67'0                                                                                X~~ error: no match found
           74:  %1 = call { ptr, i64 } @"_ZN17fn_ptr_reify_shim13DynCompatible52dyn_name_default$u7b$$u7b$reify.shim.fnptr$u7d$$u7d$17h7670877906a96512E"(ptr align 1 %_1) #2 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           75:  %_unused.01 = extractvalue { ptr, i64 } %1, 0 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           76:  %_unused.12 = extractvalue { ptr, i64 } %1, 1 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           77: ; call <fn_ptr_reify_shim::S as fn_ptr_reify_shim::NotDynCompatible>::not_dyn_name 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:67'1       ?                                                                                 possible intended match
           78:  %2 = call { ptr, i64 } @"_ZN76_$LT$fn_ptr_reify_shim..S$u20$as$u20$fn_ptr_reify_shim..NotDynCompatible$GT$12not_dyn_name17h29d9315502a38704E"() #2 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           79:  %_unused.03 = extractvalue { ptr, i64 } %2, 0 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           80:  %_unused.14 = extractvalue { ptr, i64 } %2, 1 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           81: ; call fn_ptr_reify_shim::NotDynCompatible::not_dyn_name_default 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           82:  %3 = call { ptr, i64 } @_ZN17fn_ptr_reify_shim16NotDynCompatible20not_dyn_name_default17h63f35d2e3ca4a5fdE() #2 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           83:  %_unused.05 = extractvalue { ptr, i64 } %3, 0 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           84:  %_unused.16 = extractvalue { ptr, i64 } %3, 1 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           85:  ret void 
check:67'0     ~~~~~~~~~~
           86: } 
check:67'0     ~~
           87:  
check:67'0     ~
           88: attributes #0 = { inlinehint noredzone nounwind uwtable "probe-stack"="inline-asm" "target-cpu"="generic" "target-features"="+v8a,+strict-align,+neon,+fp-armv8" } 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           89: attributes #1 = { noredzone nounwind uwtable "probe-stack"="inline-asm" "target-cpu"="generic" "target-features"="+v8a,+strict-align,+neon,+fp-armv8" } 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           90: attributes #2 = { nounwind } 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           91:  
check:67'0     ~
           92: !llvm.module.flags = !{!0} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
           93: !llvm.ident = !{!1} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~
           94:  
check:67'0     ~
           95: !0 = !{i32 4, !"kcfi", i32 1} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           96: !1 = !{!"rustc version 1.92.0-nightly (5c1ef4631 2025-09-22)"} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           97: !2 = !{i32 -1244563104} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~
           98: !3 = !{i32 -1670484703} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~
           99: !4 = !{i32 -720845901} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~
          100: !5 = !{i32 -1522505972} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~
>>>>>>

------------------------------------------

error in revision `aarch64`: verification with 'FileCheck' failed
status: exit status: 1
command: "/usr/lib/llvm-20/bin/FileCheck" "--input-file" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.aarch64/fn-ptr-reify-shim.ll" "/checkout/tests/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.rs" "--check-prefix=CHECK" "--check-prefix" "aarch64" "--allow-unused-prefixes" "--dump-input-context" "100"
stdout: none
--- stderr -------------------------------
/checkout/tests/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.rs:67:12: error: CHECK: expected string not found in input
 // CHECK: call <fn_ptr_reify_shim::S as fn_ptr_reify_shim::NotDynCompatible>::not_dyn_name{{$}}}
           ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.aarch64/fn-ptr-reify-shim.ll:73:76: note: scanning from here
; call fn_ptr_reify_shim::DynCompatible::dyn_name_default{{reify.shim.fnptr}}
                                                                           ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.aarch64/fn-ptr-reify-shim.ll:77:3: note: possible intended match here
; call <fn_ptr_reify_shim::S as fn_ptr_reify_shim::NotDynCompatible>::not_dyn_name
  ^

Input file: /checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.aarch64/fn-ptr-reify-shim.ll
Check file: /checkout/tests/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.rs

-dump-input=help explains the following input dump.

Input was:
<<<<<<
            1: ; ModuleID = 'fn_ptr_reify_shim.b6c03fc4bc8896ab-cgu.0' 
            2: source_filename = "fn_ptr_reify_shim.b6c03fc4bc8896ab-cgu.0" 
            3: target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i8:8:32-i16:16:32-i64:64-i128:128-n32:64-S128-Fn32" 
            4: target triple = "aarch64-unknown-none" 
            5:  
            6: @alloc_f5787007e3698542f22dabdab1d42ed6 = private unnamed_addr constant [11 x i8] c"dyn_default", align 1 
            7: @alloc_2172a80514eefbb0a9cb60752ddeb97d = private unnamed_addr constant [15 x i8] c"not_dyn_default", align 1 
            8: @alloc_645829e0bfec2cab47a651c7b4ba9f04 = private unnamed_addr constant [14 x i8] c"dyn_compatible", align 1 
            9: @alloc_d0a58cbf370d44771ccbd4953c6d9fdd = private unnamed_addr constant [18 x i8] c"not_dyn_compatible", align 1 
           10:  
           11: ; fn_ptr_reify_shim::DynCompatible::dyn_name_default{{reify.shim.fnptr}} 
           12: ; Function Attrs: inlinehint noredzone nounwind uwtable 
           13: define internal { ptr, i64 } @"_ZN17fn_ptr_reify_shim13DynCompatible52dyn_name_default$u7b$$u7b$reify.shim.fnptr$u7d$$u7d$17h7670877906a96512E"(ptr align 1 %_1) unnamed_addr #0 !kcfi_type !2 { 
           14: start: 
           15: ; call fn_ptr_reify_shim::DynCompatible::dyn_name_default 
           16:  %0 = call { ptr, i64 } @_ZN17fn_ptr_reify_shim13DynCompatible16dyn_name_default17h73e71e598780e0b1E(ptr align 1 %_1) #2 
           17:  %_0.0 = extractvalue { ptr, i64 } %0, 0 
           18:  %_0.1 = extractvalue { ptr, i64 } %0, 1 
           19:  %1 = insertvalue { ptr, i64 } poison, ptr %_0.0, 0 
           20:  %2 = insertvalue { ptr, i64 } %1, i64 %_0.1, 1 
           21:  ret { ptr, i64 } %2 
           22: } 
           23:  
           24: ; <fn_ptr_reify_shim::S as fn_ptr_reify_shim::DynCompatible>::dyn_name{{reify.shim.fnptr}} 
           25: ; Function Attrs: inlinehint noredzone nounwind uwtable 
           26: define internal { ptr, i64 } @"_ZN73_$LT$fn_ptr_reify_shim..S$u20$as$u20$fn_ptr_reify_shim..DynCompatible$GT$44dyn_name$u7b$$u7b$reify.shim.fnptr$u7d$$u7d$17he6409f97a7efcb24E"(ptr align 1 %_1) unnamed_addr #0 !kcfi_type !2 { 
           27: start: 
           28: ; call <fn_ptr_reify_shim::S as fn_ptr_reify_shim::DynCompatible>::dyn_name 
           29:  %0 = call { ptr, i64 } @"_ZN73_$LT$fn_ptr_reify_shim..S$u20$as$u20$fn_ptr_reify_shim..DynCompatible$GT$8dyn_name17h8a2e331ffcc7d6bbE"(ptr align 1 %_1) #2 
           30:  %_0.0 = extractvalue { ptr, i64 } %0, 0 
           31:  %_0.1 = extractvalue { ptr, i64 } %0, 1 
           32:  %1 = insertvalue { ptr, i64 } poison, ptr %_0.0, 0 
           33:  %2 = insertvalue { ptr, i64 } %1, i64 %_0.1, 1 
           34:  ret { ptr, i64 } %2 
           35: } 
           36:  
           37: ; fn_ptr_reify_shim::DynCompatible::dyn_name_default 
           38: ; Function Attrs: noredzone nounwind uwtable 
           39: define dso_local { ptr, i64 } @_ZN17fn_ptr_reify_shim13DynCompatible16dyn_name_default17h73e71e598780e0b1E(ptr align 1 %self) unnamed_addr #1 !kcfi_type !3 { 
           40: start: 
           41:  ret { ptr, i64 } { ptr @alloc_f5787007e3698542f22dabdab1d42ed6, i64 11 } 
           42: } 
           43:  
           44: ; fn_ptr_reify_shim::NotDynCompatible::not_dyn_name_default 
           45: ; Function Attrs: noredzone nounwind uwtable 
           46: define dso_local { ptr, i64 } @_ZN17fn_ptr_reify_shim16NotDynCompatible20not_dyn_name_default17h63f35d2e3ca4a5fdE() unnamed_addr #1 !kcfi_type !4 { 
           47: start: 
           48:  ret { ptr, i64 } { ptr @alloc_2172a80514eefbb0a9cb60752ddeb97d, i64 15 } 
           49: } 
           50:  
           51: ; <fn_ptr_reify_shim::S as fn_ptr_reify_shim::DynCompatible>::dyn_name 
           52: ; Function Attrs: noredzone nounwind uwtable 
           53: define dso_local { ptr, i64 } @"_ZN73_$LT$fn_ptr_reify_shim..S$u20$as$u20$fn_ptr_reify_shim..DynCompatible$GT$8dyn_name17h8a2e331ffcc7d6bbE"(ptr align 1 %self) unnamed_addr #1 !kcfi_type !3 { 
           54: start: 
           55:  ret { ptr, i64 } { ptr @alloc_645829e0bfec2cab47a651c7b4ba9f04, i64 14 } 
           56: } 
           57:  
           58: ; <fn_ptr_reify_shim::S as fn_ptr_reify_shim::NotDynCompatible>::not_dyn_name 
           59: ; Function Attrs: noredzone nounwind uwtable 
           60: define dso_local { ptr, i64 } @"_ZN76_$LT$fn_ptr_reify_shim..S$u20$as$u20$fn_ptr_reify_shim..NotDynCompatible$GT$12not_dyn_name17h29d9315502a38704E"() unnamed_addr #1 !kcfi_type !4 { 
           61: start: 
           62:  ret { ptr, i64 } { ptr @alloc_d0a58cbf370d44771ccbd4953c6d9fdd, i64 18 } 
           63: } 
           64:  
           65: ; Function Attrs: noredzone nounwind uwtable 
           66: define dso_local void @main() unnamed_addr #1 !kcfi_type !5 { 
           67: start: 
           68:  %_1 = alloca [0 x i8], align 1 
           69: ; call <fn_ptr_reify_shim::S as fn_ptr_reify_shim::DynCompatible>::dyn_name{{reify.shim.fnptr}} 
           70:  %0 = call { ptr, i64 } @"_ZN73_$LT$fn_ptr_reify_shim..S$u20$as$u20$fn_ptr_reify_shim..DynCompatible$GT$44dyn_name$u7b$$u7b$reify.shim.fnptr$u7d$$u7d$17he6409f97a7efcb24E"(ptr align 1 %_1) #2 
           71:  %_unused.0 = extractvalue { ptr, i64 } %0, 0 
           72:  %_unused.1 = extractvalue { ptr, i64 } %0, 1 
           73: ; call fn_ptr_reify_shim::DynCompatible::dyn_name_default{{reify.shim.fnptr}} 
check:67'0                                                                                X~~ error: no match found
           74:  %1 = call { ptr, i64 } @"_ZN17fn_ptr_reify_shim13DynCompatible52dyn_name_default$u7b$$u7b$reify.shim.fnptr$u7d$$u7d$17h7670877906a96512E"(ptr align 1 %_1) #2 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           75:  %_unused.01 = extractvalue { ptr, i64 } %1, 0 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           76:  %_unused.12 = extractvalue { ptr, i64 } %1, 1 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           77: ; call <fn_ptr_reify_shim::S as fn_ptr_reify_shim::NotDynCompatible>::not_dyn_name 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:67'1       ?                                                                                 possible intended match
           78:  %2 = call { ptr, i64 } @"_ZN76_$LT$fn_ptr_reify_shim..S$u20$as$u20$fn_ptr_reify_shim..NotDynCompatible$GT$12not_dyn_name17h29d9315502a38704E"() #2 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           79:  %_unused.03 = extractvalue { ptr, i64 } %2, 0 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           80:  %_unused.14 = extractvalue { ptr, i64 } %2, 1 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           81: ; call fn_ptr_reify_shim::NotDynCompatible::not_dyn_name_default 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           82:  %3 = call { ptr, i64 } @_ZN17fn_ptr_reify_shim16NotDynCompatible20not_dyn_name_default17h63f35d2e3ca4a5fdE() #2 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           83:  %_unused.05 = extractvalue { ptr, i64 } %3, 0 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           84:  %_unused.16 = extractvalue { ptr, i64 } %3, 1 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           85:  ret void 
check:67'0     ~~~~~~~~~~
           86: } 
check:67'0     ~~
           87:  
check:67'0     ~
           88: attributes #0 = { inlinehint noredzone nounwind uwtable "probe-stack"="inline-asm" "target-cpu"="generic" "target-features"="+v8a,+strict-align,+neon,+fp-armv8" } 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           89: attributes #1 = { noredzone nounwind uwtable "probe-stack"="inline-asm" "target-cpu"="generic" "target-features"="+v8a,+strict-align,+neon,+fp-armv8" } 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           90: attributes #2 = { nounwind } 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           91:  
check:67'0     ~
           92: !llvm.module.flags = !{!0} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
           93: !llvm.ident = !{!1} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~
           94:  
check:67'0     ~
           95: !0 = !{i32 4, !"kcfi", i32 1} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           96: !1 = !{!"rustc version 1.92.0-nightly (5c1ef4631 2025-09-22)"} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           97: !2 = !{i32 -1244563104} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~
           98: !3 = !{i32 -1670484703} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~
           99: !4 = !{i32 -720845901} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~
          100: !5 = !{i32 -1522505972} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~
>>>>>>
------------------------------------------

---- [codegen] tests/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.rs#aarch64 stdout end ----
---- [codegen] tests/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.rs#x86_64 stdout ----
------FileCheck stdout------------------------------

------FileCheck stderr------------------------------
/checkout/tests/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.rs:67:12: error: CHECK: expected string not found in input
 // CHECK: call <fn_ptr_reify_shim::S as fn_ptr_reify_shim::NotDynCompatible>::not_dyn_name{{$}}}
           ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.x86_64/fn-ptr-reify-shim.ll:73:76: note: scanning from here
; call fn_ptr_reify_shim::DynCompatible::dyn_name_default{{reify.shim.fnptr}}
                                                                           ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.x86_64/fn-ptr-reify-shim.ll:77:3: note: possible intended match here
; call <fn_ptr_reify_shim::S as fn_ptr_reify_shim::NotDynCompatible>::not_dyn_name
  ^

Input file: /checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.x86_64/fn-ptr-reify-shim.ll
Check file: /checkout/tests/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.rs

-dump-input=help explains the following input dump.

Input was:
<<<<<<
            1: ; ModuleID = 'fn_ptr_reify_shim.b6c03fc4bc8896ab-cgu.0' 
            2: source_filename = "fn_ptr_reify_shim.b6c03fc4bc8896ab-cgu.0" 
            3: target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-i128:128-f80:128-n8:16:32:64-S128" 
            4: target triple = "x86_64-unknown-none-elf" 
            5:  
            6: @alloc_f5787007e3698542f22dabdab1d42ed6 = private unnamed_addr constant [11 x i8] c"dyn_default", align 1 
            7: @alloc_2172a80514eefbb0a9cb60752ddeb97d = private unnamed_addr constant [15 x i8] c"not_dyn_default", align 1 
            8: @alloc_645829e0bfec2cab47a651c7b4ba9f04 = private unnamed_addr constant [14 x i8] c"dyn_compatible", align 1 
            9: @alloc_d0a58cbf370d44771ccbd4953c6d9fdd = private unnamed_addr constant [18 x i8] c"not_dyn_compatible", align 1 
           10:  
           11: ; fn_ptr_reify_shim::DynCompatible::dyn_name_default{{reify.shim.fnptr}} 
           12: ; Function Attrs: inlinehint noredzone nounwind nonlazybind uwtable 
           13: define internal { ptr, i64 } @"_ZN17fn_ptr_reify_shim13DynCompatible52dyn_name_default$u7b$$u7b$reify.shim.fnptr$u7d$$u7d$17h7670877906a96512E"(ptr align 1 %_1) unnamed_addr #0 !kcfi_type !5 { 
           14: start: 
           15: ; call fn_ptr_reify_shim::DynCompatible::dyn_name_default 
           16:  %0 = call { ptr, i64 } @_ZN17fn_ptr_reify_shim13DynCompatible16dyn_name_default17h73e71e598780e0b1E(ptr align 1 %_1) #2 
           17:  %_0.0 = extractvalue { ptr, i64 } %0, 0 
           18:  %_0.1 = extractvalue { ptr, i64 } %0, 1 
           19:  %1 = insertvalue { ptr, i64 } poison, ptr %_0.0, 0 
           20:  %2 = insertvalue { ptr, i64 } %1, i64 %_0.1, 1 
           21:  ret { ptr, i64 } %2 
           22: } 
           23:  
           24: ; <fn_ptr_reify_shim::S as fn_ptr_reify_shim::DynCompatible>::dyn_name{{reify.shim.fnptr}} 
           25: ; Function Attrs: inlinehint noredzone nounwind nonlazybind uwtable 
           26: define internal { ptr, i64 } @"_ZN73_$LT$fn_ptr_reify_shim..S$u20$as$u20$fn_ptr_reify_shim..DynCompatible$GT$44dyn_name$u7b$$u7b$reify.shim.fnptr$u7d$$u7d$17he6409f97a7efcb24E"(ptr align 1 %_1) unnamed_addr #0 !kcfi_type !5 { 
           27: start: 
           28: ; call <fn_ptr_reify_shim::S as fn_ptr_reify_shim::DynCompatible>::dyn_name 
           29:  %0 = call { ptr, i64 } @"_ZN73_$LT$fn_ptr_reify_shim..S$u20$as$u20$fn_ptr_reify_shim..DynCompatible$GT$8dyn_name17h8a2e331ffcc7d6bbE"(ptr align 1 %_1) #2 
           30:  %_0.0 = extractvalue { ptr, i64 } %0, 0 
           31:  %_0.1 = extractvalue { ptr, i64 } %0, 1 
           32:  %1 = insertvalue { ptr, i64 } poison, ptr %_0.0, 0 
           33:  %2 = insertvalue { ptr, i64 } %1, i64 %_0.1, 1 
           34:  ret { ptr, i64 } %2 
           35: } 
           36:  
           37: ; fn_ptr_reify_shim::DynCompatible::dyn_name_default 
           38: ; Function Attrs: noredzone nounwind nonlazybind uwtable 
           39: define { ptr, i64 } @_ZN17fn_ptr_reify_shim13DynCompatible16dyn_name_default17h73e71e598780e0b1E(ptr align 1 %self) unnamed_addr #1 !kcfi_type !6 { 
           40: start: 
           41:  ret { ptr, i64 } { ptr @alloc_f5787007e3698542f22dabdab1d42ed6, i64 11 } 
           42: } 
           43:  
           44: ; fn_ptr_reify_shim::NotDynCompatible::not_dyn_name_default 
           45: ; Function Attrs: noredzone nounwind nonlazybind uwtable 
           46: define { ptr, i64 } @_ZN17fn_ptr_reify_shim16NotDynCompatible20not_dyn_name_default17h63f35d2e3ca4a5fdE() unnamed_addr #1 !kcfi_type !7 { 
           47: start: 
           48:  ret { ptr, i64 } { ptr @alloc_2172a80514eefbb0a9cb60752ddeb97d, i64 15 } 
           49: } 
           50:  
           51: ; <fn_ptr_reify_shim::S as fn_ptr_reify_shim::DynCompatible>::dyn_name 
           52: ; Function Attrs: noredzone nounwind nonlazybind uwtable 
           53: define { ptr, i64 } @"_ZN73_$LT$fn_ptr_reify_shim..S$u20$as$u20$fn_ptr_reify_shim..DynCompatible$GT$8dyn_name17h8a2e331ffcc7d6bbE"(ptr align 1 %self) unnamed_addr #1 !kcfi_type !6 { 
           54: start: 
           55:  ret { ptr, i64 } { ptr @alloc_645829e0bfec2cab47a651c7b4ba9f04, i64 14 } 
           56: } 
           57:  
           58: ; <fn_ptr_reify_shim::S as fn_ptr_reify_shim::NotDynCompatible>::not_dyn_name 
           59: ; Function Attrs: noredzone nounwind nonlazybind uwtable 
           60: define { ptr, i64 } @"_ZN76_$LT$fn_ptr_reify_shim..S$u20$as$u20$fn_ptr_reify_shim..NotDynCompatible$GT$12not_dyn_name17h29d9315502a38704E"() unnamed_addr #1 !kcfi_type !7 { 
           61: start: 
           62:  ret { ptr, i64 } { ptr @alloc_d0a58cbf370d44771ccbd4953c6d9fdd, i64 18 } 
           63: } 
           64:  
           65: ; Function Attrs: noredzone nounwind nonlazybind uwtable 
           66: define void @main() unnamed_addr #1 !kcfi_type !8 { 
           67: start: 
           68:  %_1 = alloca [0 x i8], align 1 
           69: ; call <fn_ptr_reify_shim::S as fn_ptr_reify_shim::DynCompatible>::dyn_name{{reify.shim.fnptr}} 
           70:  %0 = call { ptr, i64 } @"_ZN73_$LT$fn_ptr_reify_shim..S$u20$as$u20$fn_ptr_reify_shim..DynCompatible$GT$44dyn_name$u7b$$u7b$reify.shim.fnptr$u7d$$u7d$17he6409f97a7efcb24E"(ptr align 1 %_1) #2 
           71:  %_unused.0 = extractvalue { ptr, i64 } %0, 0 
           72:  %_unused.1 = extractvalue { ptr, i64 } %0, 1 
           73: ; call fn_ptr_reify_shim::DynCompatible::dyn_name_default{{reify.shim.fnptr}} 
check:67'0                                                                                X~~ error: no match found
           74:  %1 = call { ptr, i64 } @"_ZN17fn_ptr_reify_shim13DynCompatible52dyn_name_default$u7b$$u7b$reify.shim.fnptr$u7d$$u7d$17h7670877906a96512E"(ptr align 1 %_1) #2 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           75:  %_unused.01 = extractvalue { ptr, i64 } %1, 0 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           76:  %_unused.12 = extractvalue { ptr, i64 } %1, 1 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           77: ; call <fn_ptr_reify_shim::S as fn_ptr_reify_shim::NotDynCompatible>::not_dyn_name 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:67'1       ?                                                                                 possible intended match
           78:  %2 = call { ptr, i64 } @"_ZN76_$LT$fn_ptr_reify_shim..S$u20$as$u20$fn_ptr_reify_shim..NotDynCompatible$GT$12not_dyn_name17h29d9315502a38704E"() #2 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           79:  %_unused.03 = extractvalue { ptr, i64 } %2, 0 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           80:  %_unused.14 = extractvalue { ptr, i64 } %2, 1 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           81: ; call fn_ptr_reify_shim::NotDynCompatible::not_dyn_name_default 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           82:  %3 = call { ptr, i64 } @_ZN17fn_ptr_reify_shim16NotDynCompatible20not_dyn_name_default17h63f35d2e3ca4a5fdE() #2 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           83:  %_unused.05 = extractvalue { ptr, i64 } %3, 0 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           84:  %_unused.16 = extractvalue { ptr, i64 } %3, 1 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           85:  ret void 
check:67'0     ~~~~~~~~~~
           86: } 
check:67'0     ~~
           87:  
check:67'0     ~
           88: attributes #0 = { inlinehint noredzone nounwind nonlazybind uwtable "probe-stack"="inline-asm" "target-cpu"="x86-64" "target-features"="-mmx,-sse,-sse2,-sse3,-ssse3,-sse4.1,-sse4.2,-avx,-avx2,+soft-float" } 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           89: attributes #1 = { noredzone nounwind nonlazybind uwtable "probe-stack"="inline-asm" "target-cpu"="x86-64" "target-features"="-mmx,-sse,-sse2,-sse3,-ssse3,-sse4.1,-sse4.2,-avx,-avx2,+soft-float" } 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           90: attributes #2 = { nounwind } 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           91:  
check:67'0     ~
           92: !llvm.module.flags = !{!0, !1, !2, !3} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           93: !llvm.ident = !{!4} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~
           94:  
check:67'0     ~
           95: !0 = !{i32 8, !"PIC Level", i32 2} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           96: !1 = !{i32 1, !"Code Model", i32 2} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           97: !2 = !{i32 2, !"RtLibUseGOT", i32 1} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           98: !3 = !{i32 4, !"kcfi", i32 1} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           99: !4 = !{!"rustc version 1.92.0-nightly (5c1ef4631 2025-09-22)"} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          100: !5 = !{i32 -1244563104} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~
          101: !6 = !{i32 -1670484703} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~
          102: !7 = !{i32 -720845901} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~
          103: !8 = !{i32 -1522505972} 
check:67'0     ~~~~~~~~~~~~~~~~~~~~~~~~
>>>>>>

------------------------------------------

error in revision `x86_64`: verification with 'FileCheck' failed
status: exit status: 1
command: "/usr/lib/llvm-20/bin/FileCheck" "--input-file" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.x86_64/fn-ptr-reify-shim.ll" "/checkout/tests/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.rs" "--check-prefix=CHECK" "--check-prefix" "x86_64" "--allow-unused-prefixes" "--dump-input-context" "100"
stdout: none
--- stderr -------------------------------
/checkout/tests/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.rs:67:12: error: CHECK: expected string not found in input
 // CHECK: call <fn_ptr_reify_shim::S as fn_ptr_reify_shim::NotDynCompatible>::not_dyn_name{{$}}}
           ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.x86_64/fn-ptr-reify-shim.ll:73:76: note: scanning from here
; call fn_ptr_reify_shim::DynCompatible::dyn_name_default{{reify.shim.fnptr}}
                                                                           ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.x86_64/fn-ptr-reify-shim.ll:77:3: note: possible intended match here
; call <fn_ptr_reify_shim::S as fn_ptr_reify_shim::NotDynCompatible>::not_dyn_name
  ^

Input file: /checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.x86_64/fn-ptr-reify-shim.ll
Check file: /checkout/tests/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.rs

-dump-input=help explains the following input dump.

Input was:
<<<<<<
            1: ; ModuleID = 'fn_ptr_reify_shim.b6c03fc4bc8896ab-cgu.0' 
            2: source_filename = "fn_ptr_reify_shim.b6c03fc4bc8896ab-cgu.0" 
            3: target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-i128:128-f80:128-n8:16:32:64-S128" 
            4: target triple = "x86_64-unknown-none-elf" 
            5:  
            6: @alloc_f5787007e3698542f22dabdab1d42ed6 = private unnamed_addr constant [11 x i8] c"dyn_default", align 1 
            7: @alloc_2172a80514eefbb0a9cb60752ddeb97d = private unnamed_addr constant [15 x i8] c"not_dyn_default", align 1 
            8: @alloc_645829e0bfec2cab47a651c7b4ba9f04 = private unnamed_addr constant [14 x i8] c"dyn_compatible", align 1 
```

</details>

--- a/compiler/rustc_middle/src/ty/instance.rs
+++ b/compiler/rustc_middle/src/ty/instance.rs
@@ -618,11 +618,11 @@ impl<'tcx> Instance<'tcx> {
                         // be directly reified because it's closure-like. The reify can handle the
                         // unresolved instance.
                         resolved = Instance { def: InstanceKind::ReifyShim(def_id, reason), args }
-                    // Reify `Trait::method` implementations
-                    // FIXME(maurer) only reify it if it is a vtable-safe function
+                    // Reify `Trait::method` implementations if the trait is dyn-compatible.
                     } else if let Some(assoc) = tcx.opt_associated_item(def_id)
                         && let AssocContainer::Trait | AssocContainer::TraitImpl(Ok(_)) =
                             assoc.container
+                        && tcx.is_dyn_compatible(assoc.container_id(tcx))
                     {
                         // If this function could also go in a vtable, we need to `ReifyShim` it with
                         // KCFI because it can only attach one type per function.

--- a/tests/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.rs
+++ b/tests/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.rs
@@ -1,0 +1,74 @@
+//@ add-core-stubs
+//@ revisions: aarch64 x86_64
+//@ [aarch64] compile-flags: --target aarch64-unknown-none
+//@ [aarch64] needs-llvm-components: aarch64
+//@ [x86_64] compile-flags: --target x86_64-unknown-none
+//@ [x86_64] needs-llvm-components: x86
+//@ compile-flags: -Ctarget-feature=-crt-static -Zsanitizer=kcfi -Cno-prepopulate-passes -Copt-level=0
+
+#![feature(no_core, lang_items)]
+#![crate_type = "lib"]
+#![no_core]
+
+// A `ReifyShim` should only be created when the trait is dyn-compatible.
+
+extern crate minicore;
+use minicore::*;
+
+trait DynCompatible {
+    fn dyn_name(&self) -> &'static str;
+
+    fn dyn_name_default(&self) -> &'static str {
+        let _ = self;
+        "dyn_default"
+    }
+}
+
+// Not dyn-compatible because the `Self: Sized` bound is missing.
+trait NotDynCompatible {
+    fn not_dyn_name() -> &'static str;
+
+    fn not_dyn_name_default() -> &'static str {
+        "not_dyn_default"
+    }
+}
+
+struct S;
+
+impl DynCompatible for S {
+    fn dyn_name(&self) -> &'static str {
+        "dyn_compatible"
+    }
+}
+
+impl NotDynCompatible for S {
+    fn not_dyn_name() -> &'static str {
+        "not_dyn_compatible"
+    }
+}
+
+#[no_mangle]
+pub fn main() {
+    let s = S;
+
+    // `DynCompatible` is indeed dyn-compatible.
+    let _: &dyn DynCompatible = &s;
+
+    // CHECK: call <fn_ptr_reify_shim::S as fn_ptr_reify_shim::DynCompatible>::dyn_name{{.*}}reify.shim.fnptr
+    let dyn_name = S::dyn_name as fn(&S) -> &str;
+    let _unused = dyn_name(&s);
+
+    // CHECK: call fn_ptr_reify_shim::DynCompatible::dyn_name_default{{.*}}reify.shim.fnptr
+    let dyn_name_default = S::dyn_name_default as fn(&S) -> &str;
+    let _unused = dyn_name_default(&s);
+
+    // Check using $ (end-of-line) that these calls do not contain `reify.shim.fnptr`.
+
+    // CHECK: call <fn_ptr_reify_shim::S as fn_ptr_reify_shim::NotDynCompatible>::not_dyn_name{{$}}
+    let not_dyn_name = S::not_dyn_name as fn() -> &'static str;
+    let _unused = not_dyn_name();
+
+    // CHECK: call fn_ptr_reify_shim::NotDynCompatible::not_dyn_name_default{{$}}
+    let not_dyn_name_default = S::not_dyn_name_default as fn() -> &'static str;
+    let _unused = not_dyn_name_default();
+}

--- a/tests/codegen-llvm/sanitizer/kcfi/naked-function.rs
+++ b/tests/codegen-llvm/sanitizer/kcfi/naked-function.rs
@@ -15,8 +15,9 @@ use minicore::*;
 
 struct Thing;
 trait MyTrait {
+    // NOTE: this test assumes that this trait is dyn-compatible.
     #[unsafe(naked)]
-    extern "C" fn my_naked_function() {
+    extern "C" fn my_naked_function(&self) {
         // the real function is defined
         // CHECK: .globl
         // CHECK-SAME: my_naked_function
@@ -34,13 +35,13 @@ impl MyTrait for Thing {}
 #[unsafe(no_mangle)]
 pub fn main() {
     // Trick the compiler into generating an indirect call.
-    const F: extern "C" fn() = Thing::my_naked_function;
+    const F: extern "C" fn(&Thing) = Thing::my_naked_function;
 
     // main calls the shim function
     // CHECK: call void
     // CHECK-SAME: my_naked_function
     // CHECK-SAME: reify.shim.fnptr
-    (F)();
+    (F)(&Thing);
 }
 
 // CHECK: declare !kcfi_type

--- a/tests/ui/sanitizer/kcfi-c-variadic.rs
+++ b/tests/ui/sanitizer/kcfi-c-variadic.rs
@@ -1,0 +1,20 @@
+//@ needs-sanitizer-kcfi
+//@ no-prefer-dynamic
+//@ compile-flags: -Zsanitizer=kcfi -Cpanic=abort -Cunsafe-allow-abi-mismatch=sanitizer
+//@ ignore-backends: gcc
+//@ run-pass
+
+#![feature(c_variadic)]
+
+trait Trait {
+    unsafe extern "C" fn foo(x: i32, y: i32, mut ap: ...) -> i32 {
+        x + y + ap.arg::<i32>() + ap.arg::<i32>()
+    }
+}
+
+impl Trait for i32 {}
+
+fn main() {
+    let f = i32::foo as unsafe extern "C" fn(i32, i32, ...) -> i32;
+    assert_eq!(unsafe { f(1, 2, 3, 4) }, 1 + 2 + 3 + 4);
+}

